test/integration/timeout/timeout_stack.go

```diff
@@ -0,0 +1,666 @@
+	AppContainerPort = 8080
+
+	timeoutMessage          = "upstream request timeout"
+	expectedBackendResponse = "backend"
+)
+
+// Timeout stack is setup as below
+//	FrontEnd ->
+//        - Exposes two endpoints "/default" and "/timeout" and reaches out to backend for both the endpoints
+//        - Refers to "backend" VirtualService which in turn refers to "backend" VirtualRouter
+//
+//  Backend ->
+//        - Exposes "/default" and "/timeout" with a configured 45 seconds delay
+//        - "backend" VN is configured with a Listener timeout of 60 seconds
+//        - "backend" VR has two routes "/default" and "/timeout"
+//        - "/default" path uses default timeout of 15s and "/timeout" path is configured with 60s timeout
+//
+//  We then validate the timeout feature.
+//       - Call to "/default" from frontend pod should timeout with "upstream request timeout"
+//       - Call to "/timeout" from frontend pod should get a valid response which will be "backend"(config.WHO_AM_I)
+
+type TimeoutStack struct {
+	// service discovery type
+	ServiceDiscoveryType manifest.ServiceDiscoveryType
+
+	// ====== runtime variables ======
+	mesh      *appmesh.Mesh
+	namespace *corev1.Namespace
+
+	FrontEndVN *appmesh.VirtualNode
+	FrontEndDP *appsv1.Deployment
+
+	BackEndVN  *appmesh.VirtualNode
+	BackEndDP  *appsv1.Deployment
+	BackEndSVC *corev1.Service
+
+	BackEndVS *appmesh.VirtualService
+	BackEndVR *appmesh.VirtualRouter
+}
+
+func (s *TimeoutStack) DeployTimeoutStack(ctx context.Context, f *framework.Framework) {
+	s.createTimeoutStackMeshAndNamespace(ctx, f)
+
+	s.ServiceDiscoveryType = manifest.DNSServiceDiscovery
+	mb := &manifest.ManifestBuilder{
+		Namespace:            s.namespace.Name,
+		ServiceDiscoveryType: s.ServiceDiscoveryType,
+	}
+	s.createVirtualNodeResourcesForTimeoutStack(ctx, f, mb)
+	s.createServicesForTimeoutStack(ctx, f)
+	s.assignBackendVSToFrontEndVN(ctx, f)
+}
+
+func (s *TimeoutStack) CleanupTimeoutStack(ctx context.Context, f *framework.Framework) {
+	var deletionErrors []error
+	if errs := s.revokeVirtualNodeBackendAccessForTimeoutStack(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteResourcesForTimeoutStackServices(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteResourcesForTimeoutStackNodes(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteTimeoutMeshAndNamespace(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	for _, err := range deletionErrors {
+		f.Logger.Error("clean up failed", zap.Error(err))
+	}
+	Expect(len(deletionErrors)).To(BeZero())
+}
+
+//Check Timeout behavior with and with timeout configured
+func (s *TimeoutStack) CheckTimeoutBehavior(ctx context.Context, f *framework.Framework) {
+	By(fmt.Sprintf("verify route timesout if it takes more than default 15s w/o listener timeout configured"), func() {
```

Comment:
Would be nice to make the 15 second configurable and see if Envoy sees the change - maybe enhance it later